src/FailedCheckumFilter.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Drupal\Signify;
+
+use Iterator;
+
+class FailedCheckumFilter extends \FilterIterator {
+
+    protected $workingDirectory;
+
+    public function __construct(Iterator $iterator, $working_directory) {
+        parent::__construct($iterator);
+        $this->workingDirectory = $working_directory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function accept() {
+        /** @var \Drupal\Signify\VerifierFileChecksum $checksum */
+        $checksum = $this->current();
+        $hash_file_path = $this->workingDirectory . DIRECTORY_SEPARATOR . $checksum->filename;
+        $algorithm = strtolower($checksum->algorithm);
+        $hash = @hash_file($algorithm, $hash_file_path);
```

Comment:
This is sort of where I'd hope we could use a stream wrapper instead of hashing a file path directly.

Reply:
We can work on that. I just wanted to get the general idea out there first as it seems its somewhat controversial.

Reply:
Hmm, what should be stream wrappered? I thought you were referring to construction of `ChecksumList`. But based on where this comment is placed, it seems you want to do it for every, single file name string inside of the iterator. Can you expound on your comments as we might be discussing two different things.

Reply:
To summarize our conversation in slack, this stream wrapper stuff could be done in a later follow-up.

However, to note, I don't think it would be hard to inject a stream wrapper in the mix.